inject/src/main/java/io/micronaut/inject/qualifiers/Qualifiers.java

```diff
@@ -431,6 +432,18 @@ private static <T> Qualifier<T> findCustomByName(@NonNull AnnotationMetadata met
         return null;
     }
 
+    /**
+     * Qualifies the origin bean definition that was used to create an each bean.
+     *
+     * @param beanDefinition The beanDefinition
+     * @param <T>            The component type
+     * @return The qualifier
+     * @since 4.6
+     */
+    public static <T> Qualifier<T> eachBeanOf(BeanDefinition<?> beanDefinition) {
```

Comment:
don't think we should make this public API since it quite internal specific. Either way I would name it `childOf` or `descendedFrom` something.

Reply:
Made private